tasks/gulp/__tests__/after-build-package.test.js

```diff
@@ -129,71 +132,60 @@ describe('package/', () => {
+        expect(parsedData).toBeInstanceOf(Array)
+        parsedData.forEach((option) => {
+          expect(option).toEqual(
+            expect.objectContaining({
+              name: expect.any(String),
+              type: expect.any(String),
+              required: expect.any(Boolean),
+              description: expect.any(String)
+            })
+          )
         })
+      }))
     })
   })
 
   describe('components with JavaScript', () => {
-    it.each(componentsWithJavaScript)('\'%s\' should have component JavaScript file with correct module name', (javaScriptFile) => {
-      const moduleName = componentNameToJavaScriptModuleName(path.parse(javaScriptFile).name)
-
-      return readFile(javaScriptFile, 'utf8')
-        .then((data) => {
-          expect(data).toContain("typeof define === 'function' && define.amd ? define('" + moduleName + "', factory)")
-        })
-        .catch(error => {
-          throw error
-        })
+    it('should have component JavaScript file with correct module name', () => {
+      return Promise.all(componentsWithJavaScript.map(async (componentWithJavaScript) => {
+        const data = await readFile(componentWithJavaScript, 'utf8')
+        const moduleName = componentNameToJavaScriptModuleName(path.parse(componentWithJavaScript).name)
+        expect(data).toContain("typeof define === 'function' && define.amd ? define('" + moduleName + "', factory)")
+      }))
     })
   })
 
   describe('fixtures', () => {
-    it.each(componentNames)('\'%s\' should have fixtures.json that contains JSON', (name) => {
-      const filePath = path.join(configPaths.package, 'govuk', 'components', name, 'fixtures.json')
-      return readFile(filePath, 'utf8')
-        .then((data) => {
-          const parsedData = JSON.parse(data)
-          // We expect the component JSON to contain "component" and an array of "fixtures" with "name", "options", and "html"
-          expect(parsedData).toEqual(
-            expect.objectContaining({
-              component: name,
-              fixtures: expect.any(Array)
-            })
-          )
-
-          parsedData.fixtures.forEach((fixture) => {
-            expect(fixture).toEqual({
-              name: expect.any(String),
-              options: expect.any(Object),
-              html: expect.any(String),
-              hidden: expect.any(Boolean)
-            })
+    it('should have fixtures.json that contains JSON', () => {
+      return Promise.all(components.map(async (component) => {
+        const data = await readFile(path.join(configPaths.package, 'govuk', 'components', component, 'fixtures.json'), 'utf8')
+        const parsedData = JSON.parse(data)
+
+        // We expect the component JSON to contain "component" and an array of "fixtures" with "name", "options", and "html"
+        expect(parsedData).toEqual(
+          expect.objectContaining({
+            component,
+            fixtures: expect.any(Array)
+          })
+        )
```

Comment:
I'm glad you thought this too 😊 

Yeah you do lose the separate green tick per `it.each()` _but_ because the `expect.objectContaining()` assertion has a unique `component` property (per Promise) it still prints the component name in the failure output 🙌 

Used the same approach for HTML validation too

Reply:
Neat! That's my worries sorted then 😄